packages/kit/src/runtime/server/page/render.js

```diff
@@ -80,6 +81,42 @@ export async function render_response({
 			? action_result.data ?? null
 			: null;
 
+	/** @type {string} */
+	let base = paths.base;
+
+	/** @type {string} */
+	let assets = paths.assets;
+
+	/**
+	 * An expression that will evaluate in the client to determine the resolved base path.
+	 * We use a relative path when possible to support IPFS, the internet archive, etc.
+	 */
+	let base_expression = s(paths.base);
+
+	// if appropriate, use relative paths for greater portability
+	if (paths.relative !== false && !state.prerendering?.fallback) {
+		const segments = event.url.pathname.slice(paths.base.length).split('/');
+
+		if (segments.length === 1 && paths.base !== '') {
+			// if we're on `/my-base-path`, relative links need to start `./my-base-path` rather than `.`
+			base = `./${paths.base.split('/').at(-1)}`;
+
+			base_expression = `new URL(${s(base)}, location).pathname`;
+		} else {
+			base =
+				segments
+					.slice(2)
+					.map(() => '..')
+					.join('/') || '.';
+
+			base_expression = `new URL(${s(base)}, location).pathname.slice(0, -1)`;
```

Comment:
It took me too long to understand what's going on here, an illustrative comment could help others in the future
```suggestion
			// new Url('../', location).pathname == '/foo/' -> remove trailing slash
			base_expression = `new URL(${s(base)}, location).pathname.slice(0, -1)`;
```

Reply:
```suggestion
			// resolve e.g. '../..' against current location, then remove trailing slash
			base_expression = `new URL(${s(base)}, location).pathname.slice(0, -1)`;
```